newbank/server/NewBank.java

```diff
@@ -297,6 +345,24 @@ private String payAmountToOtherCustomer(Customer customer, String request) throw
       if (amount > account.getBalance()) {
         return String.format("Insufficient balance for this transfer: %.2f", account.getBalance());
       }
+
+      //check transfer limit
+      String checkKey = userId
+          .concat(LocalDateTime.now().format(DateTimeFormatter.ISO_LOCAL_DATE).toString());
+      Double todayTransfers = transferLimit.get(checkKey);
+      if (todayTransfers == null) {
+        transferLimit.put(checkKey, amount);
+        todayTransfers = amount;
+      } else {
+        todayTransfers += amount;
+        transferLimit.put(checkKey, todayTransfers);
+      }
+      if (!SPECIAL_AUTHORISATION && todayTransfers > 10000) {
```

Comment:
I think this check should be slightly modified, so that we do not increase the transfer limit value if today's transfers exceed the limit? So in line 353, we could first check if `todayTransfers + amount > limit` and return with an error message.

Reply:
I added the code to reflect this, thanks for this input.